rest-api-spec/src/main/resources/rest-api-spec/api/async_search.submit.json

```diff
@@ -126,6 +131,16 @@
         "type":"string",
         "description":"Specify the node or shard the operation should be performed on (default: random)"
       },
+      "pre_filter_shard_size":{
```

Comment:
Could you next time please wait for a member of the team that the PR is tagged with, in this cased Search Foundations, to review it before merging?

Reply:
Sorry about that. I'll wait for a team review next time. There's indeed a test:

https://github.com/elastic/elasticsearch/blob/2ed318f21fc015609fa9b09d94115e3465c17615/x-pack/plugin/async-search/src/test/java/org/elasticsearch/xpack/search/SubmitAsyncSearchRequestTests.java#L120-L127

I opened https://github.com/elastic/elasticsearch/pull/117626 to fix it.

Reply:
That test is a unit test around the internal behaviour. What I meant is that the only way to really know that a REST parameter is supported, is to add a quick yaml test that provides it. I am assuming this would not have worked in this PR, as the parameter is not accepted.